### PR TITLE
[BUGFIX] Fix typos discovered by codespell

### PR DESCRIPTION
**Pull request by cclauss**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- Fix typos discovered by codespell
- Added a pre-commit job to detect regressions
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [ ] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [ ] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added [~unit~ tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [ ] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/checkpoint/test_simple_checkpoint.py
Comment (cdkini) [inline tests/checkpoint/test_simple_checkpoint.py:212]: This actually looks like it impacts test outcomes. Great catch!

File: .pre-commit-config.yaml
Comment (cdkini) [inline .pre-commit-config.yaml]: While I do very much like this tool, I do believe adding it to our pre-commit config is out of the scope of this current PR. 

Due to the high frequency of commits made by our developers and contributors from the GE community, I'd be wary of adding additional complexity to what is specifically designed to be a lightweight process. 

This is something that perhaps requires further discussion and thought from our developer community. If you feel strongly about this addition, could you please port it over to an issue so it may be evaluated separately from the changes made in this PR?

**Review by cdkini — CHANGES_REQUESTED** [review]
I had no idea we had so many typos so thank you for taking a look!

I think the addition of `codespell` is out of the scope of this PR (please see my comment) so once that change is reverted, I'll happily merge this in.

File: tests/checkpoint/test_simple_checkpoint.py
Comment (cclauss) [reply tests/checkpoint/test_simple_checkpoint.py:212]: Should I revert this change so the tests pass again?

Comment (cdkini) [reply tests/checkpoint/test_simple_checkpoint.py:212]: Hmm yes please! I'll add an issue/ticket around resolving this test and fix it up shortly.

**Review by cdkini — APPROVED** [review]
LGTM

**Review by cdkini — CHANGES_REQUESTED** [review]
Hmm it appears that we're still failing some integration tests after the removal of that test typo fix. The issue is pointing to Snowflake but let's dig in and see what might be causing the problem.

Additionally, your change from `faliure` to `failure` is not the cause so feel free to add it back into this PR! I'll close mine down.

**Comment by cclauss** [comment]
#3068 seems to suffer from similar snowflake failures.

**Comment by cdkini** [comment]
Hmm that's definitely quite odd. Our `docs-integration` pipeline only runs when certain files are touched so the issue may have been latent and only just surfaced after your proposed change. I don't think this is a result of a change you've made given that we're seeing a `snowflake.connector.vendored.urllib3.exceptions.NewConnectionError` in the Azure logs and you've only made changes to comments.

I believe we've dealt with this before in some capacity before so let me do some research and see what I can find.

**Comment by cdkini** [comment]
I've been working on my own PR related to docs and it appears that I'm passing these Snowflake tests.

Here's [the PR](https://github.com/great-expectations/great_expectations/pull/3053/commits/ff21272f3db3b13f3a3a20a84e6f677d65cd2c10) and here's [the Azure log](https://dev.azure.com/great-expectations/great_expectations/_build/results?buildId=11331&view=logs&j=0be527af-e1cb-5224-1725-2baa5917c313&t=096d6768-99a4-5ec0-6a65-65122b15571c). I can help look into this later but I do now think the test failures are resulting due to a change made here.

**Comment by cdkini** [comment]
@cclauss it appears that a few contributors' PR's are having this issue. I'm working to figure out what's going on and I'll hopefully have an answer for you and others sometime soon

**Comment by cdkini** [comment]
We've commented out the Snowflake tests and are resolving them internally before reactivating them. As such, you should be able to pass CI/CD and get these changes merged. Thanks!

**Comment by cclauss** [comment]
`codespell --ignore-words-list="ba,cros,fo,hist,nd,nin,onl,te,tring,wont" --skip="*.csv,*.lock"`

**Review by cdkini — APPROVED** [review]
LGTM once all tests pass!

**Pull request merged**
